src/main/java/org/simplify4u/plugins/ArtifactResolver.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2019 Slawomir Jaranowski
+ * Copyright 2019 Danny van Heumen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.simplify4u.plugins;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.artifact.resolver.ArtifactResolutionRequest;
+import org.apache.maven.artifact.resolver.ArtifactResolutionResult;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.repository.RepositorySystem;
+import org.simplify4u.plugins.skipfilters.SkipFilter;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Artifact resolver for project dependencies, build plug-ins, and build plug-in dependencies.
+ */
+final class ArtifactResolver {
+
+    private final Log log;
+
+    private final RepositorySystem repositorySystem;
+
+    private final ArtifactRepository localRepository;
+
+    private final List<ArtifactRepository> remoteRepositories;
+
+    ArtifactResolver(Log log, RepositorySystem repositorySystem, ArtifactRepository localRepository,
+            final List<ArtifactRepository> remoteRepositories) {
+        this.log = requireNonNull(log);
+        this.repositorySystem = requireNonNull(repositorySystem);
+        this.localRepository = requireNonNull(localRepository);
+        this.remoteRepositories = requireNonNull(remoteRepositories);
+    }
+
+    /**
+     * Types of dependencies: compile, provided, test, runtime, system, maven-plugin.
+     *
+     * @param filter         the artifact filter
+     * @param verifyPomFiles indicator whether to verify POM file signatures as well
+     * @return Returns set of all artifacts whose signature needs to be verified.
+     */
+    Set<Artifact> resolveProjectArtifacts(MavenProject project, SkipFilter filter, boolean verifyPomFiles)
+            throws MojoExecutionException {
+        final LinkedHashSet<Artifact> allArtifacts = new LinkedHashSet<>(
+                resolveArtifacts(project.getArtifacts(), filter, verifyPomFiles));
+        allArtifacts.addAll(resolveArtifacts(project.getPluginArtifacts(), filter, verifyPomFiles));
```

Comment:
I think that better will be pass verifyPlugins as parameter and  add plugins according to this value.

When we not always add plugin we don't need special filter to skip plugins. 

And all other "skip" filters should not have knowledge about plugin dependency.

Reply:
I know what you mean. I was doubting on this decision. From syntactical point-of-view, this makes more sense. As in: the filters work on type `Artifact`, which can be either dependency or plugin. But indeed, it does not feel like the "least surprising" solution. I've refactored.